Source/AwsCommonRuntimeKit/io/HostAddressType.swift

```diff
@@ -3,21 +3,22 @@
 
 import AwsCIo
 
-public enum AddressRecordType {
-    case typeA
-    case typeAAAA
+/// Type of Host Address (ipv4 or ipv6)
+public enum HostAddressType {
+    case A
+    case AAAA
 }
 
-extension AddressRecordType: RawRepresentable, CaseIterable {
+extension HostAddressType: RawRepresentable, CaseIterable {
 
     public init(rawValue: aws_address_record_type) {
         let value = Self.allCases.first { $0.rawValue == rawValue }
-        self = value ?? .typeA
+        self = value ?? .A
```

Comment:
not related to this PR but we should make these type of init throwable.

Falling back on .A is bad for future work, for example if we have IPv8.

Reply:
Yeah, we should refactor all enums to a better pattern.